### PR TITLE
Lyrics Plugin: DarkLyrics support

### PR DESCRIPTION
**Pull request by NoahTheDuke**
This is not ready yet, for lots of reasons, but the major one is that I'm waiting to hear back from the DarkLyrics folks about rate-limiting. The code as written works for about 15 songs, and then darklyrics blocks further requests from the same ip address for an unspecified time.

I also need to add documentation and testing, but I want to at least get this out there.

### CODE REVIEW

File: beetsplug/lyrics.py [hunk 1]
```diff
@@ -171,8 +173,15 @@ def _encode(s):
             s = s.encode('utf8', 'ignore')
         return urllib.quote(s)
 
-    def build_url(self, artist, title):
-        return self.URL_PATTERN % (self._encode(artist.title()),
+    def build_url(self, artist, album, title, url_format=None):
```
Comment (sampsyo) [inline beetsplug/lyrics.py]: Perhaps we should move the calls to `.lower()` and `.title()` to the clients of `build_url` instead of here.

**Comment by sampsyo** [comment]
Looks great! And adding `album` to the set of information passed to backends seems like a good idea too.

**Comment by NoahTheDuke** [comment]
I would suggest no one else use this, as I have gotten my current IP banned/rejected for too many requests. I have emailed the fine folks at Darklyrics about it, but they haven't responded, so who knows? Either way, watch out. :cry:

**Comment by sampsyo** [comment]
:cry:

**Pull request closed without merging**